src/upload-device.js

```diff
@@ -103,10 +107,37 @@ module.exports = {
                 fd.append(annot, classDef.annotations[annot].toJS());
         }
 
-        const dataset = await util.promisify(fs.readFile)(args.dataset, { encoding: 'utf8' });
+        const dataset = await pfs.readFile(args.dataset, { encoding: 'utf8' });
         // sanitiy check it locally
         ThingTalk.Grammar.parse(dataset);
 
+        if (args.secrets) {
+            let secretfile;
+            try {
+                secretfile = await pfs.readFile(args.secrets, { encoding: 'utf8' });
+            } catch(e) {
+                // ignore if the file does not exist, even if the argument is specified
+                // this helps with scripting through all devices
+                if (e.code !== 'ENOENT')
+                    throw e;
+                console.log('WARNING: --secrets file does not exist, ignored');
+            }
+            if (secretfile) {
+                const secrets = JSON.parse(secretfile);
```

Comment:
is it a yaml file or a json file?

Reply:
JSON.

Reply:
your help in `argparse` says "yaml". other than that, it looks good.